src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -92,6 +116,10 @@ public GoogleBucketResource lookupBucket(String bucketResourceId) {
         return bucketService.getBucketResourceById(UUID.fromString(bucketResourceId), true);
     }
 
+    public GoogleBucketResource lookupBucket(UUID bucketResourceId) {
+        return lookupBucket(bucketResourceId.toString());
```

Comment:
Since the original version creates a `UUID` from `String`, it seems it would be more efficient to move the lookup code to this method, and have the original one call this, instead of the other way around.